### PR TITLE
Add reviews endpoint

### PR DESCRIPTION
**Pull request by mikejolley**
Part of https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/658

Extents the Reviews rest API including only READ methods and checking for `edit_posts` permissions so authors have access when reading review data.

To test, ensure the block can still read reviews.

### CODE REVIEW

File: src/RestApi/ProductReviews.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * REST API Reviews controller customized for Blocks.
+ *
+ * Handles requests to the /products/reviews endpoint. These endpoints allow read-only access to editors.
+ *
+ * @internal This API is used internally by the block post editor--it is still in flux. It should not be used outside of wc-blocks.
```
Comment (Aljullu) [inline src/RestApi/ProductReviews.php]: `This API is used internally by the block post editor`

Should we update that to clarify it's also used in the frontend?

File: src/RestApi/ProductReviews.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	/**
+	 * Check if a given request has access to read the attributes.
+	 *
+	 * @param  \WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_posts' ) ) {
```
Comment (Aljullu) [inline src/RestApi/ProductReviews.php]: Non-logged-in users should have permissions to fetch reviews too, isn't it?

Comment (mikejolley) [reply src/RestApi/ProductReviews.php]: I'd like to look at that in a separate PR after checking with Claudio about security concerns.

Comment (Aljullu) [reply src/RestApi/ProductReviews.php]: Sounds good. Approving this PR, then.

**Pull request merged**